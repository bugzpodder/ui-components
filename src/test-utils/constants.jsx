// @flow
import React from "react";
import { EDC, PIPELINE } from "@grail/lib";

export const COUNTRIES: Array<Object> = [
  { label: "Afghanistan" },
  { label: "Aland Islands" },
  { label: "Albania" },
  { label: "Algeria" },
  { label: "American Samoa" },
  { label: "Andorra" },
  { label: "Angola" },
  { label: "Anguilla" },
  { label: "Antarctica" },
  { label: "Argentina" },
  { label: "Armenia" },
  { label: "Aruba" },
  { label: "Australia" },
  { label: "Austria" },
  { label: "Azerbaijan" },
  { label: "Bahamas" },
  { label: "Bahrain" },
  { label: "Bangladesh" },
  { label: "Barbados" },
  { label: "Belarus" },
  { label: "Belgium" },
  { label: "Belize" },
  { label: "Benin" },
  { label: "Bermuda" },
  { label: "Bhutan" },
].map(suggestion => ({
  label: suggestion.label,
  value: suggestion.label.toUpperCase().replace(" ", "_"),
}));

export const GRAIL = "https://grail.com/wp-content/uploads/2016/12/cropped-final-favicon-32x32.png";

export const EXAMPLE_TABLE_DATA = [
  { word: "Argle-bargle", origin: "Britain" },
  { word: "Dwaal", origin: "South Africa" },
  { word: "Floccinaucinihilipilification", origin: "General English" },
  { word: "Lollygag", origin: "North American" },
  { word: "Mouse Potato", origin: "General English" },
  { word: "Ogdoad", origin: "General English" },
  { word: "Sesquipedalian", origin: "General English" },
  { word: "Flibbertigibbet", origin: "Britain" },
  { word: "Curmudgeon", origin: "N/A" },
  { word: "Nudiustertian", origin: "Latin" },
  { word: "Tittynope", origin: "General English" },
  { word: "Pauciloquent", origin: "Britain" },
  { word: "Pneumonoultramicroscopicsilicovolcanoconiosis", origin: "General English" },
  { word: "Bijoux", origin: "Archaic" },
  { word: "Fartlek", origin: "Swedish" },
  { word: "Cruciverbalist", origin: "General English" },
  { word: "Erinaceous", origin: "General English" },
];

export const LARGE_EXAMPLE_TABLE_DATA = [];
for (let i = 0; i < 1000; i++) {
  // Create about 17,000 records.
  LARGE_EXAMPLE_TABLE_DATA.push(...EXAMPLE_TABLE_DATA);
}

export const EXTRA_LARGE_EXAMPLE_TABLE_DATA = [];
for (let i = 0; i < 100; i++) {
  // Create about 1,700,000 records
  EXTRA_LARGE_EXAMPLE_TABLE_DATA.push(...LARGE_EXAMPLE_TABLE_DATA);
}

export const ADAMS_DATA = [
  {
    date: "1848-02-23T16:20:00Z",
    content: <div>John Quincy Adams dies.</div>,
  },
  {
    date: "1818-04-10T12:00:00Z",
    content: (
      <div>
        John Quincy Adams tries to dig to the middle of the Earth to conduct trade with the people living there.
      </div>
    ),
  },
  {
    date: "1767-07-11T12:20:00Z",
    content: <div>John Quincy Adams is born.</div>,
  },
];

export const EMU_DATA = [
  {
    date: "1932-12-10T16:20:00Z",
    content: <div>War ends and is declared a victory for the emus.</div>,
  },
  {
    date: "1932-11-08T12:00:00Z",
    content: <div>Over 2500 rounds had been fired with the media claiming that only a few emus had been killed.</div>,
  },
  {
    date: "1932-11-02T12:20:00Z",
    content: <div>Sir George Pearce wages war on the local population of emus in Australia.</div>,
  },
];

export const ALPHABET_ANIMALS_DATA: Array<Array<?string>> = [
  ["alligator", "bat", "cat", "dog", "elephant", "fox"],
  ["goat", "horse", "iguana", "jaguar", "kangaroo", "lion"],
  ["monkey", "narwhal", "octopus", "penguin", "quail", "raccoon"],
  ["seal", "tiger", "umbrella bird", "vulture", "whale", "x-ray tetra"],
  ["yak", "zebra", null, null, null, null],
];

export const TEST_EXTERNAL_DOMAINS: Map<string, string> = new Map().set(EDC, "edc.com").set(PIPELINE, "pipeline.com");
