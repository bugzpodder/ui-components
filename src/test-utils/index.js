// @flow
export {
  ALPHABET_ANIMALS_DATA,
  ADAMS_DATA,
  COUNTRIES,
  EMU_DATA,
  EXTRA_LARGE_EXAMPLE_TABLE_DATA,
  LARGE_EXAMPLE_TABLE_DATA,
  EXAMPLE_TABLE_DATA,
  GRAIL,
} from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
export { StyleWrapper as ExampleWrapper } from "../style-wrapper";
