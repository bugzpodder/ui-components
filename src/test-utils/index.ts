export {
  ALPHABET_ANIMALS_DATA,
  ADAMS_DATA,
  COUNTRIES,
  COUNTRY_NAMES,
  EMU_DATA,
  EXTRA_LARGE_EXAMPLE_TABLE_DATA,
  LARGE_EXAMPLE_TABLE_DATA,
  EXAMPLE_TABLE_DATA,
  GRAIL,
  TEST_EXTERNAL_DOMAINS,
} from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
export { StyleWrapper as ExampleWrapper } from "../style-wrapper";
