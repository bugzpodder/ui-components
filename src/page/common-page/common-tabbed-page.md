### Example

```js
import { Alert } from "@material-ui/lab";
import { CommonTabbedPage } from "./";
import { ExampleBlock, ExampleWrapper } from "../test-utils";
import { useState } from "react";

const ExampleTabbedCard = () => {
  const [activeTab, setActiveTab] = useState("one");
  return (
    <>
      <CommonTabbedPage
        title="Example Tabbed Page"
        subtitle="This is a subtitle"
        activeTab={activeTab}
        onChangeActiveTab={setActiveTab}
        headerActions={[
          {
            content: "Button Two",
            color: "primary",
            onClick: () => console.debug("Button Two was clicked."),
          },
          {
            content: "Button One",
            color: "primary",
            variant: "contained",
            onClick: () => console.debug("Button One was clicked."),
          },
        ]}
        pageConfigs={[
          {
            label: "Tab One",
            key: "one",
            Component: Alert,
            componentProps: {
              color: "success",
              message: "Tab One!",
            },
          },
          {
            label: "Tab Two",
            key: "two",
            Component: Alert,
            componentProps: {
              color: "warning",
              message: "Tab Two!",
            },
          },
        ]}
      />
      <ExampleBlock strongHeader="state " content={activeTab} />
    </>
  );
};

<ExampleWrapper>
  <ExampleTabbedCard />
</ExampleWrapper>;
```

Here is an example with a side menu as well. Note that clicking on the
menu will typically cause the page to scroll to the element whose `id`
matches the `key` given in the object in `menuContents`. However, it
uses CSS anchors and these do not work on styleguidist.

```js
import { useState } from "react";
import { CommonSwitch, CommonTabbedPage } from "../../;
import { ExampleBlock, ExampleWrapper } from "../../test-utils";
import { Alert } from "@material-ui/lab";

const TabOneComponent = () => {
  return (
    <>
      <Alert severity="success" id="elementA">
        Tab One Element A!
      </Alert>
      <Alert severity="success" id="elementB">
        Tab One Element B!
      </Alert>
      <Alert severity="success" id="elementC">
        Tab One Element C!
      </Alert>
    </>
  );
};

const TabTwoComponent = () => {
  return (
    <>
      <Alert severity="warning" id="elementI">
        Tab Two Element I!
      </Alert>
      <Alert severity="warning" id="elementII">
        Tab Two Element II!
      </Alert>
      <Alert severity="warning" id="elementIII">
        Tab Two Element III!
      </Alert>
    </>
  );
};

const ExampleTabbedCard = () => {
  const [activeTab, setActiveTab] = useState("one");
  return (
    <>
      <CommonTabbedPage
        title="Example With Menu"
        subtitle="Another subtitle"
        activeTab={activeTab}
        onChangeActiveTab={setActiveTab}
        headerActions={[
          {
            Component: CommonSwitch,
            componentProps: {
              label: "CUSTOM ACTION",
              onChange: () => {},
              color: "secondary",
            },
          },
        ]}
        pageConfigs={[
          {
            label: "Tab One",
            key: "one",
            Component: TabOneComponent,
            componentProps: {},
            menuContents: [
              {
                label: "Element A",
                key: "elementA",
              },
              {
                label: "Element B",
                key: "elementB",
              },
              {
                label: "Element C",
                key: "elementC",
              },
            ],
          },
          {
            label: "Tab Two",
            key: "two",
            Component: TabTwoComponent,
            componentProps: {},
            menuContents: [
              {
                label: "Element I",
                key: "elementI",
              },
              {
                label: "Element II",
                key: "elementII",
              },
              {
                label: "Element III",
                key: "elementIII",
              },
            ],
          },
        ]}
      />
      <ExampleBlock strongHeader="state " content={activeTab} />
    </>
  );
};

<ExampleWrapper>
  <ExampleTabbedCard />
</ExampleWrapper>;
```
