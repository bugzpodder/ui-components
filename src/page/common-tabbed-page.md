### Example

```js
import { Alert } from "@grail/components";
import { ExampleBlock, ExampleWrapper } from "../test-utils";
import { Fragment, useState } from "react";
import { CommonTabbedPage } from "./";

const ExampleTabbedCard = () => {
  const [activeTab, setActiveTab] = useState("one");
  return (
    <Fragment>
      <CommonTabbedPage
        title="Example Tabbed Page"
        subtitle="This is a subtitle"
        activeTab={activeTab}
        onChangeActiveTab={setActiveTab}
        headerActions={[
          {
            content: "Button Two",
            color: "primary",
            onClick: () => console.debug("Button Two was clicked."),
          },
          {
            content: "Button One",
            color: "primary",
            variant: "contained",
            onClick: () => console.debug("Button One was clicked."),
          },
        ]}
        pageConfigs={[
          {
            label: "Tab One",
            key: "one",
            Component: Alert,
            componentProps: {
              color: "success",
              message: "Tab One!",
            },
          },
          {
            label: "Tab Two",
            key: "two",
            Component: Alert,
            componentProps: {
              color: "warning",
              message: "Tab Two!",
            },
          },
        ]}
      />
      <ExampleBlock strongHeader="state " content={activeTab} />
    </Fragment>
  );
};

<ExampleWrapper>
  <ExampleTabbedCard />
</ExampleWrapper>;
```

Here is an example with a side menu as well. Note that clicking on the
menu will typically cause the page to scroll to the element whose `id`
matches the `key` given in the object in `menuContents`. However, it
uses CSS anchors and these do not work on styleguidist.

```js
import { Fragment, useState } from "react";
import { Alert, CommonSwitch } from "@grail/components";
import { ExampleBlock, ExampleWrapper } from "../test-utils";
import { CommonTabbedPage } from "./";

const TabOneComponent = () => {
  return (
    <Fragment>
      <Alert color="success" message="Tab One Element A!" id="elementA" />
      <Alert color="success" message="Tab One Element B!" id="elementB" />
      <Alert color="success" message="Tab One Element C!" id="elementC" />
    </Fragment>
  );
};

const TabTwoComponent = () => {
  return (
    <Fragment>
      <Alert color="warning" message="Tab Two Element I!" id="elementI" />
      <Alert color="warning" message="Tab Two Element II!" id="elementII" />
      <Alert color="warning" message="Tab Two Element III!" id="elementIII" />
    </Fragment>
  );
};

const ExampleTabbedCard = () => {
  const [activeTab, setActiveTab] = useState("one");
  return (
    <Fragment>
      <CommonTabbedPage
        title="Example With Menu"
        subtitle="Another subtitle"
        activeTab={activeTab}
        onChangeActiveTab={setActiveTab}
        headerActions={[
          {
            Component: CommonSwitch,
            componentProps: {
              label: "CUSTOM ACTION",
              onChange: () => {},
              color: "secondary",
            },
          },
        ]}
        pageConfigs={[
          {
            label: "Tab One",
            key: "one",
            Component: TabOneComponent,
            componentProps: {},
            menuContents: [
              {
                label: "Element A",
                key: "elementA",
              },
              {
                label: "Element B",
                key: "elementB",
              },
              {
                label: "Element C",
                key: "elementC",
              },
            ],
          },
          {
            label: "Tab Two",
            key: "two",
            Component: TabTwoComponent,
            componentProps: {},
            menuContents: [
              {
                label: "Element I",
                key: "elementI",
              },
              {
                label: "Element II",
                key: "elementII",
              },
              {
                label: "Element III",
                key: "elementIII",
              },
            ],
          },
        ]}
      />
      <ExampleBlock strongHeader="state " content={activeTab} />
    </Fragment>
  );
};

<ExampleWrapper>
  <ExampleTabbedCard />
</ExampleWrapper>;
```
