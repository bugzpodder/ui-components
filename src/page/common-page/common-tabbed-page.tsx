import React, { ReactNode } from "react";
import classNames from "classnames";
import styles from "./common-tabbed-page.module.scss";
import { CommonPage } from "./common-page";
import {
  CommonTabbedPageClasses,
  HeaderAction,
  PageConfig,
} from "../../types/card";
import { SpinnerOverlay } from "../../spinner-overlay";
import { TabProps } from "@material-ui/core/Tab";
import { TabsComponent } from "./components/tabs-component";

type Props = {
  /**
   * The array of objects used to define the tabs displayed in the header, and their corresponding pages.
   * The `key` property is used to determine which tab is selected.
   */
  pageConfigs: PageConfig[];
  /** Takes the handler used to switch between tabs, based on the tab's value */
  onChangeActiveTab: (x0: string) => any;
  /** The value of the tab header. Used to define which tab is open */
  activeTab: string | false;
  /** Provides the page's Title */
  title?: ReactNode;
  /** Provides for action components to be rendered in the top right corner */
  headerActions?: HeaderAction[];
  /** Provides a subtitle under the `title` */
  subtitle?: ReactNode;
  /**
   * Provides classNames to the page and its sub-components. Options include:
   *
   *  - `root`
   *
   *  - `header` (applied to entire header container which contains headerActions, title, subtitle, and side menu
   *     button)
   *
   *  - `headerActions` (applied to headerActions container)
   *
   *  - `title`
   *
   *  - `subtitle`
   *
   *  - `tabs` (applied to the container of the tab buttons)
   *
   *  - `contentContainer`
   *
   *  - `content`
   *
   *  - `sideMenu`
   */
  classes?: CommonTabbedPageClasses;
  /** Props applied to the `Tabs` container */
  tabProps?: Partial<TabProps>;
  /** Displays an uninteractive loading animation over the page content when true */
  isLoading?: boolean;
};

/**
 * `CommonTabbedPage` provides a component for a page with a flush card header,
 * a tab bar, and an optional side menu. Note: if you are using this component
 * in a new environment, we recommend building a new child component for your
 * environment; take a look at the `TabbedLimsPage` component as an example.
 */
export const CommonTabbedPage: React.FC<Props> = props => {
  const {
    classes = {},
    headerActions = [],
    subtitle = "",
    title = "",
    activeTab,
    onChangeActiveTab,
    pageConfigs,
    tabProps,
    children,
    isLoading = false,
  } = props;
  const { contentContainer, ...commonPageClasses } = classes;
  if (!isLoading && !pageConfigs) {
    throw new Error("pageConfigs is required");
  }
  const selectedTab = pageConfigs.find(tab => tab.key === activeTab);
  if (!isLoading && !selectedTab) {
    console.error("no pageConfig keys match the provided activeTab");
  }
  let PageContent = null;
  let selectedMenuContents = [];
  if (selectedTab && selectedTab.Component) {
    const SelectedComponent = selectedTab.Component;
    selectedMenuContents = selectedTab.menuContents || [];
    PageContent = (
      <div className={classNames(styles.contentContainer, contentContainer)}>
        <SelectedComponent {...(selectedTab.componentProps || {})} />
      </div>
    );
  }
  return (
    <CommonPage
      classes={{ ...commonPageClasses }}
      headerActions={headerActions}
      subtitle={subtitle}
      title={title}
      menuContents={selectedMenuContents}
      subheader={
        <TabsComponent
          activeTab={!isLoading && activeTab}
          onChangeActiveTab={onChangeActiveTab}
          className={classes.tabs}
          pageConfigs={pageConfigs}
          tabProps={tabProps}
        />
      }
    >
      {PageContent}
      {children}
      <SpinnerOverlay isActive={isLoading} />
    </CommonPage>
  );
};
