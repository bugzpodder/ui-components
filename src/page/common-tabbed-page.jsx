// @flow
import React, { type Node } from "react";
import classNames from "classnames";
import styles from "./common-tabbed-page.module.scss";
import { CommonPage } from "./common-page";
import { TabsComponent } from "./components/tabs-component";

type Props = {
  /** The array of objects used to define the tabs displayed in the header, and their corresponding pages. If only
   *  one element is passed to pageConfigs, the tab bar will not be shown. */
  pageConfigs: Array<PageConfig>,
  /** Takes the handler used to switch between tabs, based on the tab's value */
  onChangeActiveTab: string => any,
  /** The value of the tab header. Used to define which tab is open */
  activeTab: string,
  /** Provides the page's Title */
  title?: Node,
  /** Provides for action components to be rendered in the top right corner */
  headerActions?: Array<HeaderAction>,
  /** Provides a subtitle under the `title` */
  subtitle?: Node,
  /**
   * Provides classNames to the page and its sub-components. Options include:
   *
   *  - `root`
   *
   *  - `header` (applied to entire header container which contains headerActions, title, subtitle, and side menu
   *     button)
   *
   *  - `headerActions` (applied to headerActions container)
   *
   *  - `title`
   *
   *  - `subtitle`
   *
   *  - `tabs` (applied to the container of the tab buttons)
   *
   *  - `contentContainer`
   *
   *  - `content`
   *
   *  - `sideMenu`
   */
  classes?: CommonTabbedPageClasses,
  /** Props applied to the `Tabs` container */
  tabProps?: Object,
};

/**
 * `CommonTabbedPage` provides a component for a page with a flush card header,
 * a tab bar, and an optional side menu. Note: if you are using this component
 * in a new environment, we recommend building a new child component for your
 * environment; take a look at the `TabbedLimsPage` component as an example.
 */
export const CommonTabbedPage = (props: Props) => {
  const {
    classes = {},
    headerActions = [],
    subtitle = "",
    title = "",
    activeTab,
    onChangeActiveTab,
    pageConfigs,
    tabProps,
  } = props;
  const { tabs, contentContainer, ...commonPageClasses } = classes;
  const selectedTab = pageConfigs.find(tab => tab.key === activeTab);
  let PageContent = null;
  let selectedMenuContents = [];
  if (selectedTab && selectedTab.Component) {
    const SelectedComponent = selectedTab.Component;
    selectedMenuContents = selectedTab.menuContents || [];
    PageContent = (
      <div className={classNames(styles.contentContainer, contentContainer)}>
        <SelectedComponent {...selectedTab.componentProps || {}} />
      </div>
    );
  }
  return (
    <CommonPage
      classes={{ ...commonPageClasses }}
      headerActions={headerActions}
      subtitle={subtitle}
      title={title}
      menuContents={selectedMenuContents}
      subheader={(
        <TabsComponent
          activeTab={activeTab}
          onChangeActiveTab={onChangeActiveTab}
          className={tabs}
          pageConfigs={pageConfigs}
          tabProps={tabProps}
        />
)}
    >
      {PageContent}
    </CommonPage>
  );
};
