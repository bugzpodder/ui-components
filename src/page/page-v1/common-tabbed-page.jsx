// @flow
import React, { type Node } from "react";
import classNames from "classnames";
import styles from "./common-tabbed-page.module.scss";
import { CommonPage } from "./common-page";
import { SpinnerOverlay } from "../../spinner-overlay";
import { TabsComponent } from "./components/tabs-component";

type Props = {
  /**
   * The array of objects used to define the tabs displayed in the header, and their corresponding pages.
   * The `key` property is used to determine which tab is selected.
   */
  pageConfigs: Array<PageConfig>,
  /** Takes the handler used to switch between tabs, based on the tab's value */
  onChangeActiveTab: string => any,
  /** The value of the tab header. Used to define which tab is open */
  activeTab: string | false,
  /** Provides the page's Title */
  title?: Node,
  /** Provides for action components to be rendered in the top right corner */
  headerActions?: Array<HeaderAction>,
  /** Provides a subtitle under the `title` */
  subtitle?: Node,
  /**
   * Provides classNames to the page and its sub-components. Options include:
   *
   *  - `root`
   *
   *  - `header` (applied to entire header container which contains headerActions, title, subtitle, and side menu
   *     button)
   *
   *  - `headerActions` (applied to headerActions container)
   *
   *  - `title`
   *
   *  - `subtitle`
   *
   *  - `tabs` (applied to the container of the tab buttons)
   *
   *  - `contentContainer`
   *
   *  - `content`
   *
   *  - `sideMenu`
   */
  classes?: CommonTabbedPageClasses,
  /** Props applied to the `Tabs` container */
  tabProps?: Object,
  /** Components to be shared between all pages */
  children?: React$Node,
  /** Displays an uninteractive loading animation over the page content when true */
  isLoading?: boolean,
};

/**
 * `CommonTabbedPage` provides a component for a page with a flush card header,
 * a tab bar, and an optional side menu. Note: if you are using this component
 * in a new environment, we recommend building a new child component for your
 * environment; take a look at the `TabbedLimsPage` component as an example.
 */
export const CommonTabbedPage = (props: Props) => {
  const {
    classes = {},
    headerActions = [],
    subtitle = "",
    title = "",
    activeTab,
    onChangeActiveTab,
    pageConfigs,
    tabProps,
    children,
    isLoading = false,
  } = props;
  const { contentContainer, ...commonPageClasses } = classes;
  if (!isLoading && !pageConfigs) {
    throw new Error("pageConfigs is required");
  }
  const selectedTab = pageConfigs.find(tab => tab.key === activeTab);
  if (!isLoading && !selectedTab) {
    console.error("no pageConfig keys match the provided activeTab");
  }
  let PageContent = null;
  let selectedMenuContents = [];
  if (selectedTab && selectedTab.Component) {
    const SelectedComponent = selectedTab.Component;
    selectedMenuContents = selectedTab.menuContents || [];
    PageContent = (
      <div className={classNames(styles.contentContainer, contentContainer)}>
        <SelectedComponent {...selectedTab.componentProps || {}} />
      </div>
    );
  }
  return (
    <CommonPage
      classes={{ ...commonPageClasses }}
      headerActions={headerActions}
      subtitle={subtitle}
      title={title}
      menuContents={selectedMenuContents}
      subheader={(
        <TabsComponent
          activeTab={!isLoading && activeTab}
          onChangeActiveTab={onChangeActiveTab}
          className={classes.tabs}
          pageConfigs={pageConfigs}
          tabProps={tabProps}
        />
)}
    >
      {PageContent}
      {children}
      <SpinnerOverlay isActive={isLoading} />
    </CommonPage>
  );
};
