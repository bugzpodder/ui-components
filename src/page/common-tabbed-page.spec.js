// @flow
import "jest-dom/extend-expect";
import React from "react";
import { Alert, CommonSwitch } from "../index";
import { CommonTabbedPage } from "./common-tabbed-page";
import { TestWrapper } from "../test-utils";
import { cleanup, fireEvent, render } from "react-testing-library";

afterEach(cleanup);

const pageConfigs: Array<PageConfig> = [
  {
    label: "Tab One",
    key: "one",
    id: "tab-one",
    Component: Alert,
    componentProps: {
      color: "success",
      message: "Tab One!",
    },
  },
  {
    label: "Tab Two",
    key: "two",
    id: "tab-two",
    Component: Alert,
    componentProps: {
      color: "warning",
      message: "Tab Two!",
    },
  },
];

const headerActions = [
  {
    content: "Test Action",
  },
  {
    Component: CommonSwitch,
    componentProps: {
      label: "CUSTOM ACTION",
      onChange: () => {},
    },
  },
];

test("render common tabbed page", () => {
  const mockOnChange = jest.fn(result => result);
  const { container, getByTestId } = render(
    <TestWrapper>
      <CommonTabbedPage
        title="Test Card"
        activeTab="one"
        pageConfigs={pageConfigs}
        headerActions={headerActions}
        onChangeActiveTab={mockOnChange}
      />
    </TestWrapper>,
  );
  expect(getByTestId("alert")).toHaveTextContent("Tab One!");
  expect(getByTestId("tab-one")).toHaveAttribute("aria-selected", "true");
  expect(getByTestId("alert")).not.toHaveTextContent("Tab Two!");
  expect(getByTestId("tab-two")).toHaveAttribute("aria-selected", "false");
  fireEvent.click(getByTestId("tab-two"));
  expect(mockOnChange.mock.results[0].value).toEqual("two");
  expect(container).toMatchSnapshot();
});

test("render empty common tabbed page", () => {
  const { container } = render(
    <TestWrapper>
      <CommonTabbedPage
        pageConfigs={[]}
        activeTab=""
        onChangeActiveTab={() => {}}
      />
    </TestWrapper>,
  );
  expect(container).toMatchSnapshot();
});
