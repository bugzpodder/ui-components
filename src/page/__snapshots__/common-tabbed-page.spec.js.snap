// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render common tabbed page 1`] = `
<div>
  <div
    class="pageContainer"
    data-testid="common-page"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded MuiCard-root stickyHeader pageHeader"
    >
      <div
        class="MuiCardHeader-root cardHeader"
        data-testid="common-page-card-header"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title header-title"
          >
            Test Card
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiCardHeader-subheader"
          />
        </div>
        <div
          class="MuiCardHeader-action headerActions"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-flat"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Test Action
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="commonSwitch"
          >
            <label
              class="MuiFormControlLabel-root controlLabel"
              data-testid="common-switch"
            >
              <span
                class="MuiSwitch-root"
              >
                <span
                  class="MuiButtonBase-root MuiIconButton-root MuiSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <span
                      class="MuiSwitch-icon"
                    />
                    <input
                      class="MuiSwitchBase-input"
                      id=""
                      type="checkbox"
                      value=""
                    />
                  </span>
                </span>
                <span
                  class="MuiSwitch-bar"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body2 MuiFormControlLabel-label"
              >
                CUSTOM ACTION
              </span>
            </label>
            
          </div>
        </div>
      </div>
      <div
        class="MuiTabs-root"
        data-testid="card-tabs"
      >
        <div
          class="MuiTabs-flexContainer"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed"
            role="tablist"
            style="margin-bottom: 0px;"
          >
            <div
              class="MuiTabs-flexContainer"
            >
              <button
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-selected"
                data-testid="tab-one"
                id="tab-one"
                role="tab"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiTab-wrapper"
                >
                  <span
                    class="MuiTab-labelContainer"
                  >
                    <span
                      class="MuiTab-label"
                    >
                      Tab One
                    </span>
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary"
                data-testid="tab-two"
                id="tab-two"
                role="tab"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiTab-wrapper"
                >
                  <span
                    class="MuiTab-labelContainer"
                  >
                    <span
                      class="MuiTab-label"
                    >
                      Tab Two
                    </span>
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <span
              class="MuiTabIndicator-root MuiTabIndicator-colorPrimary MuiTabs-indicator"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="contentAndMenu"
    >
      <div
        class="content"
      >
        <div
          class="contentContainer"
        >
          <div
            class="alertContainer"
            data-testid="alert"
          >
            <div
              class="MuiTypography-root MuiTypography-body2 MuiPaper-root MuiPaper-elevation6 MuiSnackbarContent-root success alert-success"
              role="alertdialog"
            >
              <div
                class="MuiSnackbarContent-message"
              >
                <span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root alertIcon"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                  Tab One!
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`render empty common tabbed page 1`] = `
<div>
  <div
    class="pageContainer"
    data-testid="common-page"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded MuiCard-root stickyHeader pageHeader"
    >
      <div
        class="MuiCardHeader-root cardHeader"
        data-testid="common-page-card-header"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title header-title"
          />
          <span
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiCardHeader-subheader"
          />
        </div>
        <div
          class="MuiCardHeader-action headerActions"
        />
      </div>
      <div
        class="MuiTabs-root"
        data-testid="card-tabs"
      >
        <div
          class="MuiTabs-flexContainer"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed"
            role="tablist"
            style="margin-bottom: 0px;"
          >
            <div
              class="MuiTabs-flexContainer"
            />
            <span
              class="MuiTabIndicator-root MuiTabIndicator-colorPrimary MuiTabs-indicator"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="contentAndMenu"
    >
      <div
        class="content"
      />
    </div>
  </div>
</div>
`;
